learn/quickstart/derived_types.md

```diff
@@ -0,0 +1,298 @@
+
+```fortran
+type :: my_type
+    integer :: i
+    real    :: x
+end type
+```
+
+The syntax for creating a variable of type _my_type_ and accessing its members:
+```fortran
+! declare
+type(my_type) :: foo
+! initialize
+foo%i   = 1
+foo%x   = 0.5
+```
+
+{% include note.html content="In Fortran the percentage symbol `%` is used to access the members of a derived type." %}
+
+To initialize the members of _my_type_ one can use either individual initialization as demonstrated in the above example, or the assignment operator (=), or the default initialization.
+
+
+Example using the assignement operator (=):
+```fortran
+foo = my_type(1, 0.5)
+! or using F2003 stardard and on
+foo = my_type(i=1, x=0.5)
+```
+
+Example with default initialization:
+```fortran
+type :: my_type
+    integer :: i = 1
+    real    :: x = 0.5
+end type
+! then it is possible to use as
+type(my_type) :: foo
+foo = my_type(i=2) ! foo%i gets a new value, but foo%x retains the default one.
+```
+
+## Derived types in detail
+
+The full syntax of a derived type with all optional properties is presented below:
+
+```fortran
+type [,attribute-list] :: name [(parameterized-decleration-list)]
+    [parameterized-definition-statements]
+    [private statement or sequence statement]
+    [member-variables]
+    contains
+        [type-bound-procedures]
+end type
+```
+
+### Part 1: Options to declare a derived type
+
+`attribute-list` may refer to the following:
+
+- _access-type_ that is either `public` or `private`
+- `bind(c)` offers interoperability with C programming language 
+- `extends(`_parent_`)` where _parent_ is the name of a previously declared derived type, from which, the current derived type will inherit all its members and functionality.
+- `abstract` an object orianted feature that is covered in the advanced programming tutorial.
+
+{% include note.html content="If the `attribute: bind(c)` or the `statement: sequence` is used then a derived type cannot have the `attribute: extends` and visa-versa." %}
+
+The `sequence` attribute may be used only to declare that the following  members should be accessed in the same order as they are defined within the derived type. 
+
+Example with `sequence`:
+```fortran
+type :: foo
+sequence
+integer :: var1
+real    :: var2
+end type
+! init
+type(foo) :: bar
+bar = foo(1, 0.5)
+```
+{% include note.html content="The use of statement `sequence` presupposes that the data types defined below are neither of `allocatable` nor of `pointer` type. Furthermore, it does not imply that these data types  will be stored in memory in any particular form, there is no relation to `contigeous` attribute." %}
+
+The _access-type_ attributes `public` and `private` if used, declare that all [member-variables] declared below will be automatically assigned the attribute accordingly. 
+
+The attribute `bind(c)` is used to achieve compatibility between Fortran's derived type and C's struct.
+
+Example with `bind(c)`:
+```fortran
+module mymod
```

Comment:
I would remove this line, or a `end module mymod` should be added in the example